src/Aspire.Hosting.Azure.ApplicationInsights/AzureApplicationInsightsExtensions.cs

```diff
@@ -93,20 +93,12 @@ public static IResourceBuilder<AzureApplicationInsightsResource> AddAzureApplica
                         // side and the CDK side so that AZD can fill the value in with the one it generates.
                         appInsights.WorkspaceResourceId = autoInjectedLogAnalyticsWorkspace.Id;
                     }
-                    else
-                    {
-                        // If the user does not supply a log analytics workspace of their own, and we are in publish mode
-                        // then we want AZD to provide one to us.
-                        var logAnalyticsWorkspaceParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.LogAnalyticsWorkspaceId, typeof(string));
-                        infrastructure.Add(logAnalyticsWorkspaceParameter);
-                        appInsights.WorkspaceResourceId = logAnalyticsWorkspaceParameter;
```

Comment:
We'll need to document how to get this feature back. We might want to support a model where an appinsights resource can be attached to any resource that implement `IAzureLogAnalyticsWorkspace` (which ACA would implement).